internal/lcow/disk.go

```diff
@@ -0,0 +1,55 @@
+package lcow
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/sirupsen/logrus"
+)
+
+// FormatDisk creates a utility vm, mounts the disk as a scsi disk onto to the VM
+// and then formats it with ext4. Disk is expected to be made offline before this
+// command is run. The following powershell commands:
+// 'Get-Disk -Number <disk num> | Set-Disk -IsOffline $true'
+// can be used to offline the disk.
+func FormatDisk(ctx context.Context, lcowUVM *uvm.UtilityVM, destPath string) error {
+	removeSCSI := false
+	if lcowUVM == nil {
+		return fmt.Errorf("no uvm")
+	}
+
+	if lcowUVM.OS() != "linux" {
+		return errors.New("lcow::FormatDisk requires a linux utility VM to operate")
+	}
+
+	log.G(ctx).WithFields(logrus.Fields{
+		"dest": destPath,
+	}).Debug("lcow::FormatDisk opts")
+
+	scsi, err := lcowUVM.AddSCSIPhysicalDisk(ctx, destPath, "", false) // No destination as not formatted
+	if err != nil {
+		return err
+	}
+	removeSCSI = true
+	defer func() {
+		if removeSCSI {
+			scsi.Release(ctx)
+		}
+	}()
```

Comment:
Still not understanding this as if we reached this point then removeSCSI is always true so we'll always run the condition in the defer. What's the reason for the variable?

Reply:
Apologies! Removed the unnecessary check.